code/game/machinery/machinery.dm

```diff
@@ -328,11 +327,12 @@ Class Procs:
 	if((machine_flags & FIXED2WORK) && !anchored)
 		return FALSE
 
-	var/area/this_area = get_area(src)
-	if(!this_area)
-		return FALSE
+	if(this_area)
+		return this_area.powered(chan)
+	else
+		var/area/power_area = get_area(src)
+		return power_area ? power_area.powered(chan) : FALSE
```

Comment:
It's not *strictly* equivalent, but you can just do `power_area?.powered(chan)` instead of the ternary